systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java

```diff
@@ -4,101 +4,222 @@
  */
 package io.strimzi.systemtest.logs;
 
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.fabric8.kubernetes.api.model.ContainerStatus;
+import io.fabric8.kubernetes.api.model.Pod;
 import io.strimzi.systemtest.Constants;
 import io.strimzi.test.k8s.KubeClient;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.TimeZone;
 
 import static io.strimzi.test.TestUtils.writeFile;
 import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
 
-public class LogCollector {
+/**
+ * LogCollector collects logs from all resources **if test case or preparation phase fails**. All can be found
+ * inside ./systemtest/target/logs directory where the structure of the logs are as follows:
+ *
+ * ./logs
+ *      /time/
+ *          /test-suite_time/
+ *              /test-case/
+ *                  /configmaps/
+ *                  /events
+ *                  deployment.log
+ *                  configmap.log
+ *                  ...
+ *
+ *              cluster-operator.log    // shared cluster operator logs for all tests inside one test suite
+ *          /another-test-suite_time/
+ *      ...
+ */
+public class LogCollector implements LogCollect {
     private static final Logger LOGGER = LogManager.getLogger(LogCollector.class);
 
-    private KubeClient kubeClient;
-    private String namespace;
-    private File logDir;
-    private File configMapDir;
-    private File eventsDir;
+    private static final String CURRENT_DATE;
+
+    static {
+        // Get current date to create a unique folder
+        final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyyMMdd_HHmmss");
```

Comment:
I think we should use same date pattern as we use in STs and pods (`yyyy-MM-dd HH:mm:ss` ?)